lib/galaxy/datatypes/data.py

```diff
@@ -430,14 +431,16 @@ def _serve_raw(
         headers["content-type"] = (
             "application/octet-stream"  # force octet-stream so Safari doesn't append mime extensions to filename
         )
-        filename = self._download_filename(
+        filename, utf8_encoded_filename = self._download_filename(
             dataset,
             to_ext,
             hdca=kwd.get("hdca"),
             element_identifier=kwd.get("element_identifier"),
             filename_pattern=kwd.get("filename_pattern"),
         )
-        headers["Content-Disposition"] = f'attachment; filename="{filename}"'
+        headers["Content-Disposition"] = (
+            f"attachment; filename=\"{filename}\"; filename*=UTF-8''{utf8_encoded_filename}"
```

Comment:
Can we move this to a utility function to reduce duplication?
Something like `def toContentDisposition(filename: str)` or something similar. The utf8 encoding should be inside too, so `_download_filename` just returns `filename`.

Reply:
Thanks for the suggestion! Should I remove the `escape` from `_download_filename`?

Reply:
If it is not relevant anymore I guess so? it seems to be used in `hdca_name` so check carefully if you can really drop it :+1:

Reply:
I dropped the `escape` function and also removed it from `hdca_name` too, as it didn’t seem necessary since the quote function handles it effectively.

Reply:
The escape function limits input to 150 characters and excludes characters that aren't file system safe.

Reply:
So we have to keep this but I will move this part into the new function.